### ISSUE TITLE
Patrones: Tarea 02

### ISSUE THREAD

**Issue by domingomery** [domingomery/patrones#21]
Por favor hacer comentarios aquí. Respondo todas las preguntas siempre y cuando no las haya aclarado en clases, o no estén escritas en el enunciado de la Tarea, o no las haya contestado en este foro. Usemos el foro responsablemente para que sea de utilidad. Preguntas repetidas, pueden hacer este foro excesivamente largo, lo que dificultará su lectura. Muchas gracias! Domingo

**Comment by VicenteMerino** [comment]
Profe, respecto a clean, cuando tengo características que son invariantes a la rotación y a la escala, estos debo inmediatamente eliminarlos? O debo quedarme con una de ellos?

**Comment by domingomery** [comment]
La selección manual se hace antes del clean. En la selección manual debes decidir las características que se pueden extraer para evitar falsas correlaciones. En la tarea habría que pensar, dado que las caras están todas de frente y bien alineadas, si estaría permitido sacara características invariantes a la rotación.

**Comment by fnberrios** [comment]
Hola profe, tengo una duda. Si realizo la extracción de características de LBP, pero divido la imagen en 9 sectores, ¿en realidad estoy sacando 9x59 características no? Me refiero a si esas 9x59 cuentan dentro de las 200 características mínimas que tenemos que obtener. Gracias de antemano.

**Comment by IchottMano** [comment]
Hola!

Podemos reusar código de las actividades para la tarea?

Muchas gracias.

**Comment by porellanau** [comment]
Hola profesor! En la etapa de obtener las características mi código demora aproximadamente 8.5 minutos, así que guardé tanto la matriz de características como los _labels_ en archivos `.npy`. Al hacer la entrega, ¿basta subir tanto el archivo `.ipynb` junto con los anteriormente mencionados a _classroom_ o hay una forma de que _Colab_ guarde estos archivos extras para hacer la entrega?

Muchas gracias de antemano.

**Comment by domingomery** [comment]
> Hola profe, tengo una duda. Si realizo la extracción de características de LBP, pero divido la imagen en 9 sectores, ¿en realidad estoy sacando 9x59 características no? Me refiero a si esas 9x59 cuentan dentro de las 200 características mínimas que tenemos que obtener. Gracias de antemano.

Con esto cumples lo del enunciado, no hay duda, y está bien. Sin embargo, la idea que yo tenía era que probaran con distintas familias de características, es decir no sólo con LBP. No es obligatorio obviamente.

**Comment by domingomery** [comment]
> Podemos reusar código de las actividades para la tarea?

Sí, se puede, pero hay que tener en cuenta las restricciones del enunciado (porque no todos los métodos que están en los Colab cumplen con estas restricciones).

**Comment by domingomery** [comment]
> En la etapa de obtener las características mi código demora aproximadamente 8.5 minutos, así que guardé tanto la matriz de características como los _labels_ en archivos `.npy`. Al hacer la entrega, ¿basta subir tanto el archivo `.ipynb` junto con los anteriormente mencionados a _classroom_ o hay una forma de que _Colab_ guarde estos archivos extras para hacer la entrega?

Puedes hacerlo de la forma que señalas. Por favor usa nombres de archivo que empiecen con tu número de alumno (esto porque al final todas las tareas se guardan en un solo directorio del Google Drive y se arma un enredo en las correcciones).

**Comment by fnberrios** [comment]
> > Hola profe, tengo una duda. Si realizo la extracción de características de LBP, pero divido la imagen en 9 sectores, ¿en realidad estoy sacando 9x59 características no? Me refiero a si esas 9x59 cuentan dentro de las 200 características mínimas que tenemos que obtener. Gracias de antemano.
> 
> Con esto cumples lo del enunciado, no hay duda, y está bien. Sin embargo, la idea que yo tenía era que probaran con distintas familias de características, es decir no sólo con LBP. No es obligatorio obviamente.

Gracias profe, lo tendré en cuenta.

**Comment by IchottMano** [comment]
> > Podemos reusar código de las actividades para la tarea?
> 
> Sí, se puede, pero hay que tener en cuenta las restricciones del enunciado (porque no todos los métodos que están en los Colab cumplen con estas restricciones).

Muchas gracias profe! Sobre esto, es válido usar PLSRegression? Estaba con la duda si se podía usar o no.

**Comment by domingomery** [comment]
> Muchas gracias profe! Sobre esto, es válido usar PLSRegression? Estaba con la duda si se podía usar o no.

Prefiero que uds. mismos se contesten este tipo de preguntas, así pueden saber mejor cómo funcionan los métodos que están usando :)

**Comment by fguinez** [comment]
> > En la etapa de obtener las características mi código demora aproximadamente 8.5 minutos, así que guardé tanto la matriz de características como los _labels_ en archivos `.npy`. Al hacer la entrega, ¿basta subir tanto el archivo `.ipynb` junto con los anteriormente mencionados a _classroom_ o hay una forma de que _Colab_ guarde estos archivos extras para hacer la entrega?
> 
> Puedes hacerlo de la forma que señalas. Por favor usa nombres de archivo que empiecen con tu número de alumno (esto porque al final todas las tareas se guardan en un solo directorio del Google Drive y se arma un enredo en las correcciones).

Hola profesor, en el caso de usar archivos `.npy` añadimos de todas formas el código que genera los vectores de características en el Colab o solo explicamos cómo se generaron los vectores?

**Comment by domingomery** [comment]
> Hola profesor, en el caso de usar archivos `.npy` añadimos de todas formas el código que genera los vectores de características en el Colab?

Así es, la idea es que tu código pueda ejecutarse de dos maneras, una en la que se calculen todas las características y otra en la que se lean las características extraídas desde los archivos.

>  ... o solo explicamos cómo se generaron los vectores?

No, debes incluir también el código de cómo se extraen las características.

**Comment by bjbarias** [comment]
Hola Profe! 
Cual es el tiempo máximo de ejecución?
Saludos

**Comment by domingomery** [comment]
> Cual es el tiempo máximo de ejecución?

Ideal no más de 10 minutos... si se demora mucho por favor poner también el archivo de las características extraídas (por respeto al tiempo de quienes hacen la corrección).

**Comment by tpcantergiani** [comment]
Hola profesor!
Se puede ocupar la funcion mlxsfs de la libreria mlxtend?
Me entro la duda, ya que buscando la libreria note que era de machine learning.

Gracias de antemano

**Comment by domingomery** [comment]
> Se puede ocupar la funcion mlxsfs de la libreria mlxtend?
> Me entro la duda, ya que buscando la libreria note que era de machine learning.

No manejo el detalle, pero hay que tener cuidado con la función objetivo (de separabilidad) que maximiza la librería. Hay muchos metodos que miden la separabilidad entrenando una red neuronal, clasificando y calculando el accuracy (porque el accuracy es una muy buena medida de la separabilidad). En este caso, no está permitido usar así la librería... pero puede ser que exista una opción en la que la separabilidad se mida sin tener que entrenar. Te pediría que veas bien el detalle de que hace la librería y que opciones estás usando.

**Comment by tpcantergiani** [comment]
Gracias Profesor!
Preferí eliminarla finalmente.
Otra pregunta la funcion para realizar PLSR que usa en algunos ejemplos.
`from   sklearn.cross_decomposition import PLSRegression`

Se considera como una técnica de deep learning, ya que se comporta parecido a un modelo de redes.

Saludos,

**Comment by tpcantergiani** [comment]
> > Muchas gracias profe! Sobre esto, es válido usar PLSRegression? Estaba con la duda si se podía usar o no.
> 
> Prefiero que uds. mismos se contesten este tipo de preguntas, así pueden saber mejor cómo funcionan los métodos que están usando :)

Gracias por la pronta respuesta profe,

Con respecto a la funcion para implementar PLSR:
`from   sklearn.cross_decomposition import PLSRegression`

Esta permitida usar esta implementación?

Leí bien la libreria, y no me queda claro si el algoritmo `Nipals` se considera como técnica de Deep Learning, o como técnica de clasificación distinta a KKN.

Gracias de antemano

**Comment by domingomery** [comment]
> Con respecto a la funcion para implementar PLSR: `from sklearn.cross_decomposition import PLSRegression` Esta permitida usar esta implementación?cLeí bien la libreria, y no me queda claro si el algoritmo `Nipals` se considera como técnica de Deep Learning, o como técnica de clasificación distinta a KKN.

Como dije anteriormente, prefiero que sean uds. los que contesten este tipo de preguntas. Si entiendes qué hace `Nipals` podrás ver si es una técnica de clasificación, o una técnica de deep learning.

**Comment by xDanman** [comment]
Hola profesor! Tengo una consulta con respecto a lo que no se puede usar, citando del enunciado:

> Ningún proceso de la tarea debe tener algoritmos de clasificación basados en técnicas distintas al KNN, es decir no se puede usar un selector o transformador de características basado en redes neuronales o SVM por ejemplo.

Esto quiere decir que las técnicas de selección basados en arboles, que coloco en `PAT03_SeleccionGeneral`, no serian utilizables dado que utilizan clasificadores diferentes a KNN de 3 vecinos? como lo es `LogisticRegression`, `RandomForestClassifier` y `LGBMClassifier`.

**Comment by domingomery** [comment]
> Esto quiere decir que las técnicas de selección basados en arboles, que coloco en `PAT03_SeleccionGeneral`, no serian utilizables dado que utilizan clasificadores diferentes a KNN de 3 vecinos? como lo es `LogisticRegression`, `RandomForestClassifier` y `LGBMClassifier`.

Correcto, estos métodos no se pueden usar.
